src/kvirc/ui/KviUserListView.cpp

```diff
@@ -485,60 +487,115 @@ void KviUserListView::completeNickBashLike(const QString & szBegin, std::vector<
 	}
 }
 
-bool KviUserListView::completeNickStandard(const QString & szBegin, const QString & szSkipAfter, QString & szBuffer, bool bAppendMask)
+bool KviUserListView::completeNickStandard(const QString & szBegin, QString & szSkipAfter, QString & szBuffer, bool bAppendMask)
 {
-	KviUserListEntry * pEntry = m_pHeadItem;
+	// For normal nickname completion, we'll build a list of matching nicknames in order,
+	// then look through it to the next nickname each time Tab is pressed.
+	if (m_CompletionList == nullptr)
+	{
+		// No list; build it.
+		m_CompletionList = new std::list<QString>();
+
+		KviUserListEntry * pEntry = m_pHeadItem;
+
+		for(KviUserListEntry * pEntry = m_pHeadItem; pEntry; pEntry = pEntry->m_pNext)
+		{
+			// FIXME: completion should skip my own nick or place it as last entry in the chain (?)
+
+			//	if(KviConsoleWindow * c = m_pKviWindow->console())
+			//	{
+			//		if(kvi_strEqualCI(entry->m_szNick.ptr(),c->currentNickName())
+			//	}
+
+			if(pEntry->m_szNick.length() >= szBegin.length())
+			{
+				bool bEqual = KviQString::equalCIN(szBegin, pEntry->m_szNick, szBegin.length());
+				if(!bEqual && KVI_OPTION_BOOL(KviOption_boolIgnoreSpecialCharactersInNickCompletion))
+				{
+					QString szTmp = pEntry->m_szNick;
+					szTmp.remove(QRegExp("[^a-zA-Z0-9]"));
+					bEqual = KviQString::equalCIN(szBegin, szTmp, szBegin.length());
+				}
+
+				if(bEqual)
+				{
+					// Matching entry; insert it into the list.
+					std::list<QString>::iterator it;
+					switch (KVI_OPTION_UINT(KviOption_uintNickCompletionOrder))
+					{
+						case 0:  // As listed
+							m_CompletionList->push_back(pEntry->m_szNick);
+							break;
+						case 1:  // Alphabetical
+							for (it = m_CompletionList->begin(); it != m_CompletionList->end(); ++it)
+							{
+								if ((KviQString::cmpCI(pEntry->m_szNick, *it,
```

Comment:
Extra set of parentheses are only needed with assignment in the conditional.  Also, extra space between `if` and `(...)`.